### PR TITLE
feat(odyssey-react): improve Icon types

### PR DESCRIPTION
**Pull request by ghost**
add additional JSDOC for components without `children`

### CODE REVIEW

**Comment by jeffbelser-okta** [comment]
@andrewberg-okta can you give this one a look today? thank you

File: packages/odyssey-react/src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -23,6 +23,9 @@ export interface CheckboxProps
     ComponentPropsWithRef<"input">,
     "onChange" | "style" | "className" | "type" | "children"
   > {
+  /**
+   * Children are never rendered.
+   */
   children?: never;
```
Comment (arnoldsandoval-okta) [inline packages/odyssey-react/src/components/Checkbox/Checkbox.tsx:29]: If this is the case, why include this at all?

Comment (jeffbelser-okta) [reply packages/odyssey-react/src/components/Checkbox/Checkbox.tsx:29]: good question - I would say its a decision born out of our larger API design of providing "pass through" props to underlying DOM nodes as much as possible. There really is little harm in providing `children` when it isn't used but I thought we could add some finesse to components in our offering when they don't render children to make that clear at authorship time through TS language server integrations and even document it on our storybook site.

File: packages/odyssey-icons/.svgo.yml [hunk 2]
```diff
@@ -58,8 +58,7 @@ plugins:
   - removeAttrs: false
   - removeAttributesBySelector: false
   - removeElementsByAttr: false
-  - addClassesToSVGElement:
-      classNames: ["ods-icon"]
```
Comment (andrewberg-okta) [inline packages/odyssey-icons/.svgo.yml:62]: I don't think we should remove this yet. The stand alone icons provided by this package are available to be used in other contexts. Without this class (and the accompanying style) they won't display correctly and will need to be edited individually by consuming teams.

```
.ods-icon {
  position: relative;
  inset-block-start: -0.05em;
  width: 1em;
  height: 1em;
  vertical-align: middle;
}
```
specifically the class is responsible for the correct width and height.

File: packages/odyssey-react/src/components/Icon/SvgIcon.tsx [hunk 3]
```diff
@@ -36,28 +42,17 @@ export interface SvgIconProps
...
 
     return Children.only(
-      cloneElement(
-        children,
-        {
-          ...omitProps,
-          "aria-labelledby": title && autoId,
-          className: styles.root,
-          ref: ref,
-          role: title ? "img" : "presentation",
```
Comment (andrewberg-okta) [inline packages/odyssey-react/src/components/Icon/SvgIcon.tsx:50]: Either will work but "presentaion" feels more correct in this case and is (despite the storybook error) a valid role

Comment (andrewberg-okta) [reply packages/odyssey-react/src/components/Icon/SvgIcon.tsx:50]: I don't believe the "img" role needs to be set explicitly, if not specified it should fall back to its default of "graphics-document" which should be sufficient

**Pull request merged**